src/libra_py/cube_file_methods.py

```diff
@@ -0,0 +1,263 @@
+
+
+
+
+def read_volumetric_data(filename: str):
+    """
+    This function reads the volumetric data in a format used for plotting
+    of the data. The difference between 'read_cube' function is that it will
+    show the data in a 3D array which has the shape as the number of grid points
+    for each of the X-, Y-, and Z-axis in the 4th to 6th line of the cube file.
+	
+    This function will return the grid points in each axis, the coordinates of the
+    structure and the spacing vector which is used to plot the isosurfaces of the 
+    molecular orbitals.
+	
+    Args:
+	
+        filename (string): The name of the .cube file.
+		
+    Returns:
+	
+        coordinates (2D numpy array): The coordinates of the molecule structure in
+                                      the same format shown in the .cube file.
+		
+        X_3d, Y_3d, Z_3d (3D numpy array): Containing the grid points for each of the X-,
+                                     Y-, and Z- axis.
+		
+        V_3d (3D numpy array): The volumetric data in a 3D numpy array format.
+		
+        
+	spacing_vector (numpy 1D array): The spacing vector used for plotting the isosurfaces.
+
+    """
+
+    f = open(filename,'r')
+    lines = f.readlines()
+    f.close()
+	
+    # The number of atoms in the 3rd line
+    natoms = int(lines[2].split()[0])
+
+
+    # The number of voxels defined for each axis obtained from
+    # the first elements of the 4th, 5th, and 6th line of the cube files
+    nx = int(lines[3].split()[0])
+    ny = int(lines[4].split()[0])
+    nz = int(lines[5].split()[0])
+
+
+    # The three vectors below are the same vectors which are present in lines 4th to 6th
+    # which are used to create the 3D numpy array of the grid points (x, y, z) used to plot
+    # the isosurfaces.
+    # Here we use the same unit as is used in the .cube file structure which is Bohr
+    # with no need for unit conversion. This makes the plotting easier.
+    axis_1 = np.array([float(lines[3].split()[1]),float(lines[3].split()[2]),float(lines[3].split()[3])])
+    axis_2 = np.array([float(lines[4].split()[1]),float(lines[4].split()[2]),float(lines[4].split()[3])])
+    axis_3 = np.array([float(lines[5].split()[1]),float(lines[5].split()[2]),float(lines[5].split()[3])])
+
+
+    # The spacing vector. This will be used in the 'marching_cubes_lewiner'
+    # function to define the vertices and faces for 'plot_trisurf' function.
+    # This vector is obtained from the sum of the three axis in the cube file as above.
+    spacing_vector = axis_1+axis_2+axis_3
+    
+
+    # First we read all the isovalues into a 1D list 'isovals'.
+    isovals = []
+
+	
+    # Starting from the line which the volumetric data starts which is the (natoms+3+2+1+1)th line.
+    for i in range(natoms+3+2+1,len(lines)):
+        for j in range(0,len(lines[i].split())):
+            isovals.append(float(lines[i].split()[j]))
+    
+
+    # Define the volumetric numpy array
+    V_3d = np.zeros((nx,ny,nz))

```

Comment:
better name like "v_3d", but maybe even better - is there any more descriptive name?